SalahAdDin/strapi-plugin-rich-text
# feat: #10 add settings to the plugin

### What does it do?

It adds settings routes, configuration, services, and controllers for loading default and saved settings.

### Why is it needed?

Content managers want to customize the editor toolbar and features, so we give them a mechanism for doing that.

### How to test it?
Add the desired settings on your plugin notification(plugin.ts):
```ts
export default {
  "richtext-editor": {
    enabled: true,
    resolve: "./node_modules/strapi-plugin-rich-text",
    config: {
      headings: ["h2", "h3", "h4", "h4", "h5", "h6"],
      other: {
        wordcount: true,
        saveJson: true,
      },
    },
  },
};
```

Run the project and go to the content type, choose any of the created entities, and check the editor toolbar.

You can go to settings and look for RichText Editor settings, check your desired settings on the page, modify them, save them, and check on the content type entities that your changes are reflected.

### Notes
- Loading settings duplicate the fetching times by two, and there are already to fetches there.
![Screenshot_20240830_185857](https://github.com/user-attachments/assets/dc7a00eb-909a-4742-a034-38a00f568b0f)
![Screenshot_20240830_185948](https://github.com/user-attachments/assets/68a0fd6c-2a7f-4efb-9890-36c706196086)
- Adding a loading component was required since returning a null value when loading breaks the rendering.
![Screenshot_20240830_190001](https://github.com/user-attachments/assets/65e01d1b-7416-4edd-94c4-aec0b1aae363)
- Config validation could be handled by using any library like [Valibot](https://valibot.dev/).
